sphinxgallery/gen_rst.py

```diff
@@ -310,81 +339,57 @@ def execute_script(image_dir, thumb_file, image_fname, base_image_name,
     if os.path.exists(time_path):
         time_elapsed = float(open(time_path).read())
 
-    if not os.path.exists(first_image_file) or \
-       os.stat(first_image_file).st_mtime <= os.stat(src_file).st_mtime:
-        # We need to execute the code
-        print('plotting %s' % fname)
-        t0 = time()
-        import matplotlib.pyplot as plt
```

Comment:
But I think it misses the point of not requiring matplotlib if we call it anyway to set the backend Agg before everything.